cmd/endpoint.go

```diff
@@ -235,6 +207,8 @@ func NewEndpointList(args ...string) (endpoints EndpointList, err error) {
 			return nil, fmt.Errorf("mixed style endpoints are not supported")
 		} else if endpoint.Scheme != scheme {
 			return nil, fmt.Errorf("mixed scheme is not supported")
+		} else if endpoint.Scheme == "https" && scheme == "https" && !globalIsSSL {
```

Comment:
endpoint should not worry about any globals.  This check should be outside of endpoint creation.